### PR TITLE
Update name of template output dir flag

### PR DESCRIPTION
**Pull request by nmuesch**
### Requirements for Contributing to this repository

* Fill out the template below. Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
* The pull request must only fix one issue, or add one feature, at the time.
* The pull request must update the test suite to demonstrate the changed functionality.
* After you create the pull request, all status checks must be pass before a maintainer reviews your contribution. For more details, please see [CONTRIBUTING](/CONTRIBUTING.md).

### What does this PR do?

Update the name of the output-dir flag to more explicitly note that it'll be the output directory of the generated templates, not the output directory of the generate command. 

### Description of the Change

The name of this flag was confusing me, especially when passing it into the generate command. The `output-dir` to me, meant the output directory of the generation step, not the templates. 

Changing this name makes that much more clear. 

### Possible Drawbacks

This will require any users of the existing flag to update to the new flag name. However, when using `container-apigentools` there are defaults that don't require these flags, so they shouldn't be currently relied on too much. 

### Verification Process

I ran apigentools generate locally with this new flag and confirmed the directory I passed in contained the pulled down templates. 

### Review checklist (to be filled by reviewers)

- [ ] Feature or bug fix MUST have appropriate tests (unit, integration, etc...)
- [ ] PR title must be written as a CHANGELOG entry [(see why)](/CONTRIBUTING.md#pull-request-title)
- [ ] Files changes must correspond to the primary purpose of the PR as described in the title (small unrelated changes should have their own PR)
- [ ] PR must have one `changelog/` label attached. If applicable it should have the `backward-incompatible` label attached.
- [ ] PR should not have `do-not-merge/` label attached.
- [ ] If Applicable, issue must have `kind/` and `severity/` labels attached at least.

### CODE REVIEW

File: docs/cli.md [hunk 1]
```diff
@@ -124,7 +124,7 @@ Obtains upstream `openapi-generator` templates, applies template patches, and sa
 Argument | Description | Environment Variable | Default
 ---------|-------------|----------------------|--------
 `-h, --help` | Show help message and exit.
-`-o OUTPUT_DIR, --output-dir OUTPUT_DIR` | Path to directory where processed upstream templates are saved. | `APIGENTOOLS_TEMPLATES_DIR` | `templates`
+`-to TEMPLATES_OUTPUT_DIR, --templates-output-dir TEMPLATES_OUTPUT_DIR` | Path to directory where processed upstream templates are saved. | `APIGENTOOLS_TEMPLATES_DIR` | `templates`
```
Comment (jirikuncar) [inline docs/cli.md]: I am not a big fan of short options with more than one character.

File: docs/cli.md [hunk 2]
```diff
@@ -74,7 +74,7 @@ Since version 0.10.0, apigentools also supports processing templates (equivalent
 Argument | Description | Environment Variable | Default
 ---------|-------------|----------------------|--------
 `--templates-source {local-dir,openapi-git,openapi-jar,skip}` | Source to use for obtaining templates to be processed (`skip` to not process templates) | `APIGENTOOLS_TEMPLATES_SOURCE`| `skip`
-`-o OUTPUT_DIR, --output-dir OUTPUT_DIR` | Path to directory where processed upstream templates are saved. | `APIGENTOOLS_TEMPLATES_DIR` | `templates`
+`-to TEMPLATES_OUTPUT_DIR, --templates-output-dir TEMPLATES_OUTPUT_DIR` | Path to directory where processed upstream templates are saved. | `APIGENTOOLS_TEMPLATES_DIR` | `templates`
```
Comment (jirikuncar) [inline docs/cli.md]:
```suggestion
`-T TEMPLATES_OUTPUT_DIR, --templates-output-dir TEMPLATES_OUTPUT_DIR` | Path to directory where processed upstream templates are saved. | `APIGENTOOLS_TEMPLATES_DIR` | `templates`
```

File: docs/cli.md (hunk 1 above)
Comment (jirikuncar) [inline docs/cli.md]:
```suggestion
`-T TEMPLATES_OUTPUT_DIR, --templates-output-dir TEMPLATES_OUTPUT_DIR` | Path to directory where processed upstream templates are saved. | `APIGENTOOLS_TEMPLATES_DIR` | `templates`
```

**Comment by nmuesch** [comment]
Sounds good. I wasn't sure what would be a good name for this field 😆

File: apigentools/cli.py [hunk 3]
```diff
@@ -83,7 +83,7 @@
     },
 )
 TEMPLATES_OUTPUT_DIR_ARG = (
-    ("-o", "--output-dir"),
+    ("-to", "--templates-output-dir"),
```
Comment (jirikuncar) [inline apigentools/cli.py]:
```suggestion
    ("-T", "--templates-output-dir"),
```

**Pull request merged**